web/src/compose.js

```diff
@@ -169,7 +178,137 @@ export function send_message_success(request, data) {
     }
 }
 
-export function send_message(request = create_message_object()) {
+let split_long_messages_banner_shown = false;
+
+export function update_split_long_messages_info_banner() {
+    if (check_need_to_split_long_messages()) {
+        show_split_long_messages_info_banner();
+        compose_ui.autosize_textarea($("textarea#compose-textarea"));
+    } else {
+        clear_split_long_messages_info_banner();
+    }
+}
+
+const SPLIT_DELIMITER = "\n\n\n";
+/**
+ * checks whether the message actually needs to be split, as it is possible that `split_long_messages` is enabled but the message doesn't contain the delimiter.
+ */
+function check_need_to_split_long_messages() {
+    if (!is_split_long_messages_enabled()) {
+        return false;
+    }
+    const content = compose_state.message_content();
+    return content.includes(SPLIT_DELIMITER);
+}
+
+export function is_split_long_messages_enabled() {
+    return localStorage.getItem("split_long_messages") === "true";
+}
+
+function show_split_long_messages_info_banner() {
+    if (split_long_messages_banner_shown) {
+        return;
+    }
+    const context = {
+        banner_type: compose_banner.INFO,
+        banner_text: $t({
+            defaultMessage: `Your message will be split into different parts.`,
+        }),
+        button_text: $t({
+            defaultMessage: `Send as a single message`,
+        }),
+        classname: compose_banner.CLASSNAMES.split_long_messages,
+    };
+    const new_row_html = render_compose_banner(context);
+    compose_banner.append_compose_banner_to_banner_list($(new_row_html), $("#compose_banners"));
+    split_long_messages_banner_shown = true;
+}
+
+function clear_split_long_messages_info_banner() {
+    if (!split_long_messages_banner_shown) {
+        return;
+    }
+    $(`#compose_banners .${CSS.escape(compose_banner.CLASSNAMES.split_long_messages)}`).remove();
+    split_long_messages_banner_shown = false;
+}
+
+export function toggle_split_long_messages() {
+    let state = is_split_long_messages_enabled();
+    if (state) {
+        localStorage.setItem("split_long_messages", "false");
+    } else {
+        localStorage.setItem("split_long_messages", "true");
+    }
+    state = !state;
+
+    // preview area and compose banners should be updated with the new setting
+    if (is_preview_area_showing()) {
+        // update the preview to show/hide horizontal lines showing where the content will be split
+        show_preview_area();
+    }
+    update_split_long_messages_info_banner();
+    const $toggle_button = $(".split_long_messages_toggle");
+    if ($toggle_button.length === 0) {
+        return;
+    }
+    $toggle_button.text(
+        $t(
+            {defaultMessage: "{split_long_messages_toggle_button_text}"},
+            {
+                split_long_messages_toggle_button_text: state
+                    ? "Send as a single message"
+                    : "Send as multiple messages",
```

Comment:
Translation tags with logic inside don't work at all. 

See https://zulip.readthedocs.io/en/latest/translating/internationalization.html to get some general understanding of how translation works, but in general all your if/else clauses need to be outside translation tags in order for the translation library to find strings. You could also consider just using a handlebars template to render the string rather than having JS edit it; that might be a lot cleaner.

Reply:
I'm not sure you need logic like this at all; doesn't `web/templates/popovers/send_later_popover.hbs` get rerendered every time it's opened?

Reply:
It does, but I wasn't closing the popover when the button was pressed, so I had to manually sync the state.
I'll implement that and remove this logic.